File: project.py
import networkx as nx
import matplotlib.pyplot as plt

# Create a directed graph
G = nx.DiGraph()

# Add nodes with positions
G.add_node('A', pos=(0, 0))
G.add_node('B', pos=(2, 2))
G.add_node('C', pos=(2, -2))
G.add_node('D', pos=(4, 0))

# Add edges with weights
G.add_edge('A', 'B', weight=4, capacity=10)
G.add_edge('A', 'C', weight=2, capacity=5)
G.add_edge('B', 'C', weight=5, capacity=15)
G.add_edge('B', 'D', weight=10, capacity=10)
G.add_edge('C', 'D', weight=3, capacity=10)

# Set node positions
pos = nx.get_node_attributes(G, 'pos')

# Source and target nodes
source_node = 'A'
target_node = 'D'

# Function to find the shortest path using Dijkstra's algorithm
def shortest_path(graph, source, target):
    shortest_path = nx.shortest_path(graph, source=source, target=target, weight='weight')
    shortest_distance = nx.shortest_path_length(graph, source=source, target=target, weight='weight')
    return shortest_path, shortest_distance

# Function to find the maximum flow using Ford-Fulkerson algorithm
def maximum_flow(graph, source, target):
    flow_value, flow_dict = nx.maximum_flow(graph, source, target)
    return flow_value, flow_dict

# Find the shortest path
shortest_path, shortest_distance = shortest_path(G, source_node, target_node)
print(f"Shortest Path: {shortest_path}")
print(f"Shortest Distance: {shortest_distance}")

# Find the maximum flow
flow_value, flow_dict = maximum_flow(G, source_node, target_node)
print(f"Maximum Flow Value: {flow_value}")
print("Flow Dictionary:")
for u, flows in flow_dict.items():
    for v, flow in flows.items():
        print(f"Flow from {u} to {v}: {flow}")

# Plot the graph with coordinates
plt.figure(figsize=(8, 6))
nx.draw(G, pos, with_labels=True, node_size=3000, node_color='lightblue')
labels = nx.get_edge_attributes(G, 'weight')
nx.draw_networkx_edge_labels(G, pos, edge_labels=labels)
plt.title('Graph with Coordinates')
plt.show()

File: project.js
const NetworkX = require('networkx-js');
const nx = new NetworkX();

// Function to find the shortest path using Dijkstra's algorithm
function shortestPath(graph, source, target) {
  const shortestPath = nx.shortestPath(graph, source, target, 'weight');
  const shortestDistance = nx.shortestPathLength(graph, source, target, 'weight');
  return [shortestPath, shortestDistance];
}

// Function to find the maximum flow using Ford-Fulkerson algorithm
function maximumFlow(graph, source, target) {
  const flowValue = nx.maximumFlow(graph, source, target);
  return flowValue;
}

// Create a directed graph
const G = new nx.DiGraph();

// Add edges with weights and coordinates
G.addEdge('A', 'B', { weight: 4, capacity: 10, pos: { A: [0, 0], B: [2, 2] } });
G.addEdge('A', 'C', { weight: 2, capacity: 5, pos: { A: [0, 0], C: [2, -2] } });
G.addEdge('B', 'C', { weight: 5, capacity: 15, pos: { B: [2, 2], C: [2, -2] } });
G.addEdge('B', 'D', { weight: 10, capacity: 10, pos: { B: [2, 2], D: [4, 0] } });
G.addEdge('C', 'D', { weight: 3, capacity: 10, pos: { C: [2, -2], D: [4, 0] } });

// Source and target nodes
const sourceNode = 'A';
const targetNode = 'D';

// Find the shortest path
const [shortestPath, shortestDistance] = shortestPath(G, sourceNode, targetNode);
console.log(`Shortest Path: ${shortestPath}`);
console.log(`Shortest Distance: ${shortestDistance}`);

// Find the maximum flow
const flowValue = maximumFlow(G, sourceNode, targetNode);
console.log(`Maximum Flow Value: ${flow
